### PR TITLE
Added docker configuration and pihole to smartdns converter

### PR DESCRIPTION
**Pull request by SamTV12345**
- Adds a docker configuration to get started with smartdns-rs with Docker.
- Adds a pihole to smartdns converter

The first one solves the problem that I want to host smartdns not on a host and rather in docker-compose/K3S. It is configured by mapping a conf file into the container. This can be done in Kubernetes e.g. with a initContainer

As far as I can see there are only 2 "official" adlists for smartdns. This converter script enables the user to convert a pihole adlist which always has the format 0.0.0.0 address to address /address/# which does not resolve in smartdns. 
The output in smartdns-ads.txt can then be copied into your custom smartdns.conf file which then in return can be mounted into the Docker container.

### CODE REVIEW

**Comment by mokeyish** [comment]
Thanks for the contributions. 

Can you move the docker related to the `contrib/docker` directory? 

Because these changes include a specific version number, it is unlikely that I will update it every release.  Put it in contrib, so that other users who need it can refer to it.

**Comment by mokeyish** [comment]
`address /address/#` returns SOA

`address /address/0.0.0.0` returns 0.0.0.0


Which one is better depends on your choice.

**Comment by SamTV12345** [comment]
I'll try to change that to 0.0.0.0 and check if that solves the problem that some dns entries can escape the blocklist.

**Comment by mokeyish** [comment]
> Concerning the docker part you could add that to the release pipeline. You can override the ARG default value with --args. An example is given [here](https://docs.docker.com/build/guide/build-args/). After the release is published you can trigger that docker build pipeline and push it to your own Dockerhub/Github account registry.

Let's move over there first. In fact, docker is not necessary for this project. The program itself is also single-file binary, easy to deploy, and only reads one configuration file. For better performance, docker is not recommended.

**Comment by SamTV12345** [comment]
> A more convenient way is putting domain names at a indenpendent file and load it by `domain-set`.
> 
> domains.txt
> 
> ```
> a.com
> www.google.com
> ```
> 
> smartdns.conf
> 
> ```ini
> # load `domains.txt` as `abc`
> domain-set -name abc -f domains.txt
> 
> # then
> address /domain-set:abc/# 
> # or
> address /domain-set:abc/0.0.0.0
> ```

Good idea. I'll do that.

**Comment by SamTV12345** [comment]
I did change the directory and decided for 0.0.0.0. With that a domain is 100% blocked.

**Comment by SamTV12345** [comment]
Hope this is okay now.

File: etc/pihole-to-smartdns.sh
Comment (mokeyish) [inline etc/pihole-to-smartdns.sh:1]: A very useful script, also moved to contrib.

File: etc/.gitignore
Comment (mokeyish) [inline etc/.gitignore:1]: This is? Should it be placed in contrib/docker?

**Review by mokeyish — CHANGES_REQUESTED** [review]

**Comment by SamTV12345** [comment]
I moved everything to contrib/docker.

**Review by mokeyish — APPROVED** [review]

**Pull request merged**